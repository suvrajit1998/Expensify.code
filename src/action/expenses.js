import uuid from "uuid";

export const addexpense = ({
  decription = "",
  note = "",
  amount = 0,
  createdAt = 0
}) => ({
  type: "ADD_EXPENSES",
  expenses: {
    id: uuid(),
    decription,
    note,
    amount,
    createdAt
  }
});

export const removeid = ({ id } = {}) => ({
  type: "REMOVE_ID",
  id
});

export const editexpense = (id, updets) => ({
  type: "EDIT_EXPENSE",
  id,
  updets
});
