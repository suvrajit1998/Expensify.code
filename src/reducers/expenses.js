const expensreducerdefaultarray = [];

export default (state = expensreducerdefaultarray, action) => {
  switch (action.type) {
    case "ADD_EXPENSES": {
      return [...state, action.expenses];
    }
    case "REMOVE_ID": {
      return state.filter(({ id }) => id !== action.id);
    }
    case "EDIT_EXPENSE": {
      return state.map(expenses => {
        if (expenses.id === action.id) {
          return {
            ...expenses,
            ...action.updets
          };
        } else {
          return expenses;
        }
      });
    }
    default:
      return state;
  }
};
