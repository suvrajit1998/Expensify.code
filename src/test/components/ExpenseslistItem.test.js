import React from "react";
import { shallow } from "enzyme";
import ExpensesItem from "../../components/expenselistitem";
import expenses from "../fixture/expenses";

test("Should render ExpenseListItem with Expenses", () => {
  const wrapper = shallow(<ExpensesItem {...expenses[1]} />);

  expect(wrapper).toMatchSnapshot();
});
