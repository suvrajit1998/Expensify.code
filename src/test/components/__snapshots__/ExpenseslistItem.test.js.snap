// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseListItem with Expenses 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/2"
  >
    <h3>
      rent
    </h3>
  </Link>
  <p>
    195
     - 
    -345600000
  </p>
</div>
`;
